src/components/Popups/PopupCursor.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import { FormattedMessage } from 'react-intl';
+import messages from './messages';
+import PopupBase from './PopupBase';
+import { Options } from './Popper';
+import './PopupCursor.scss';
+
+export type Props = {
+    isActive: boolean;
+};
+
+const options: Partial<Options> = {
+    modifiers: [
+        {
+            name: 'offset',
+            options: {
+                offset: [0, 20],
+            },
+        },
+    ],
+    placement: 'right',
+};
+
+const generateGetBoundingClientRect = (x = 0, y = 0) => () => ({
+    width: 0,
+    height: 0,
+    top: y,
+    right: x,
+    bottom: y,
+    left: x,
+});
+
+export default class PopupCursor extends React.PureComponent<Props> {
+    popupRef = React.createRef<PopupBase>();
+
+    isDuplicated = false;
+
+    virtualElement = {
+        getBoundingClientRect: generateGetBoundingClientRect(),
+    };
+
+    constructor(props: Props) {
+        super(props);
+        if (document.querySelector('.ba-PopupCursor')) {
+            this.isDuplicated = true;
+        } else {
+            document.addEventListener('mousemove', this.updatePosition);
+        }
+    }
+
+    componentWillUnmount(): void {
+        if (!this.isDuplicated) {
+            document.removeEventListener('mousemove', this.updatePosition);
+        }
+    }
+
+    getCreatorRects = (): DOMRect[] => {
+        const rects: DOMRect[] = [];
+        document.querySelectorAll('.ba-RegionAnnotations-creator').forEach(creator => {
+            rects.push(creator.getBoundingClientRect());
+        });
+        return rects;
+    };
+
+    isInRange = (x: number, y: number): boolean =>
+        this.getCreatorRects().reduce(
```

Comment:
This seems expensive to do -- query the document for the creator element on each mousemove event. Could we pass in either the element we want to restrict this to or maybe even just the rect as a prop?